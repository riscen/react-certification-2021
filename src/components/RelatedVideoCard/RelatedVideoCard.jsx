import React from 'react';
import { Link } from 'react-router-dom';
import { Paper } from '@material-ui/core';

import {
  RelatedVideo,
  RelatedVideoThumbnail,
  RelatedVideoTitle,
} from './RelatedVideoCard.modules';
import './RelatedVideoCard.styles.css';

function RelatedVideoCard(props) {
  const { videoId, title, imageUrl } = props;

  return (
    <Link to={`/video/${videoId}`} key={videoId}>
      <Paper className="related-video-card" elevation={3}>
        <RelatedVideo>
          <RelatedVideoThumbnail>
            <img alt={title} src={imageUrl} />
          </RelatedVideoThumbnail>
          <RelatedVideoTitle>{title}</RelatedVideoTitle>
        </RelatedVideo>
      </Paper>
    </Link>
  );
}

export default RelatedVideoCard;
