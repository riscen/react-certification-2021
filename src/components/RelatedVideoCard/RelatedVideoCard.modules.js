import styled from 'styled-components';

export const RelatedVideo = styled.div`
  display: flex;
  flex-direction: row;
  margin: 5px 0 0 0;
  width: 100%;
`;

export const RelatedVideoThumbnail = styled.div`
  margin-right: 5px;
`;

export const RelatedVideoTitle = styled.div`
  font-size: 18px;
  font-weight: normal;
  display: -webkit-box;
  text-overflow: ellipsis;
  overflow: hidden;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
`;
