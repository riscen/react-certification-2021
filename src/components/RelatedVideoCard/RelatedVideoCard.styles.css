.related-video-card {
  overflow: hidden;
  margin-bottom: 10px;
  height: 100px;
}
