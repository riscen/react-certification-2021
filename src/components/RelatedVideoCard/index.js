export { default } from './RelatedVideoCard';
export {
  RelatedVideo,
  RelatedVideoThumbnail,
  RelatedVideoTitle,
} from './RelatedVideoCard.modules';
