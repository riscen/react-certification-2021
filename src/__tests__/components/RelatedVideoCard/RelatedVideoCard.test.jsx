import React from 'react';
import { BrowserRouter, Link } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
import { Paper } from '@material-ui/core';

import RelatedVideoCard, {
  RelatedVideo,
  RelatedVideoThumbnail,
  RelatedVideoTitle,
} from '../../../components/RelatedVideoCard';

describe('RelatedVideoCard', () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  it('Renders VideoDetails Page', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RelatedVideoCard />
      </BrowserRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders Link', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RelatedVideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(Link)).toBeDefined();
  });

  it('Contains Correct Link', () => {
    const id = 'id';
    const otherId = 'otherId';
    const wrapper = mount(
      <BrowserRouter>
        <RelatedVideoCard videoId={id} />
      </BrowserRouter>
    );
    expect(wrapper.find(`a[href="/video/${id}"]`).exists()).toBeTruthy();
    expect(wrapper.find(`a[href="/video/${otherId}"]`).exists()).toBeFalsy();
  });

  it('Renders Related Video Card Container', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RelatedVideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(Paper)).toBeDefined();
  });

  it('Renders RelatedVideo', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RelatedVideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(RelatedVideo)).toBeDefined();
  });

  it('Renders RelatedVideoThumbnail', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RelatedVideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(RelatedVideoThumbnail)).toBeDefined();
  });

  it('Contains img', () => {
    const title = 'title';
    const imageUrl = 'img.url';
    const wrapper = mount(
      <BrowserRouter>
        <RelatedVideoCard title={title} imageUrl={imageUrl} />
      </BrowserRouter>
    );
    expect(wrapper.find(`img[alt="${title}"]`).exists()).toBeTruthy();
    expect(wrapper.find(`img[src="${imageUrl}"]`).exists()).toBeTruthy();
  });

  it('Renders RelatedVideoTitle', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RelatedVideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(RelatedVideoTitle)).toBeDefined();
  });
});
