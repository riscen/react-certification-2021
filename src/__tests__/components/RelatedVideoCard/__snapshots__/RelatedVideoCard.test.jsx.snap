// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedVideoCard Renders VideoDetails Page 1`] = `ReactWrapper {}`;
